common/main/java/com/couchbase/lite/internal/core/C4QueryObserver.java

```diff
@@ -137,6 +137,8 @@ void queryChanged() {
         if ((results != null) || (err != null)) { callback.onQueryChanged(results, err); }
     }
 
+    // ??? This is called only from LiteCore.
+    // Should it still be guarded by the db lock?
```

Comment:
I checked iOS code and iOS doesn't lock the database when calling `c4queryobs_getEnumerator`.

Reply:
Jens says it need to be done...

> These functions follow the general rule: only one thread at a time can call any of these functions with a specific C4Database reference.
> ...
> Queries
>    ...
>     c4queryobs_* (except c4queryobs_create which is thread-safe)